SymFloat: Expose comparison operators in C++ API

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #94073
* __->__ #94812

This is adapted from the corresponding methods in `SymInt.h`.

peterbell10 committed:
SymFloat: Expose comparison operators in C++ API

[ghstack-poisoned]

Skylion007 commented on c10/core/SymFloat.cpp:
```diff
@@ -62,6 +62,69 @@ SymFloat SymFloat::operator/(const SymFloat& sci) const {
   return SymFloat(res[0]->truediv(res[1]));
 }
 
+SymBool SymFloat::sym_eq(const SymFloat& sci) const {
```
Why not just overload the operators in C++ properly since we can do that in C++? Is it because we return a SymBool type?

Skylion007 replied:
Ah nvm, just saw it in the header.

peterbell10 committed:
Update on "SymFloat: Expose comparison operators in C++ API"


This is adapted from the corresponding methods in `SymInt.h`.

[ghstack-poisoned]
